Doc: update term "namespace package" (#129251)
python/cpython: Doc/glossary.rst

      modules, respectively.

   namespace package

>       A :pep:`420` :term:`package` which serves only as a container for

Reviewer: Why do you remove the mention of PEP 420? It is correct in it being part of PEP 420?

I don’t see why any changes to this section are necessary, the other change is just formatting.

Author: I don't remove it. I just moved it.
It is because "A PEP 420 package" is not friendly to readers.

      Namespace packages may have no physical representation,
      and specifically are not like a :term:`regular package` because they
      have no ``__init__.py`` file.

>       Namespace packages allow you to split single package across multiple directories or distributions.

Reviewer: ```suggestion
      Namespace packages allow you to split a single package across multiple directories or distributions.
```
Small typo

>       Namespace packages allow you to split a single package across multiple directories or distributions.

Author: ```suggestion
      Namespace packages allow several individually installable packages to have a common parent package.
```

>       On the other hand, namespace packages have some drawbacks and pitfalls.

Reviewer: ```suggestion
      On the other hand, namespace packages have several drawbacks and pitfalls.
```

I think „such as” and an example or two would be good here.

Author: It is difficult to explain in one sentence.

Explanations for those who need namespace packages can be found at packaging.python.org.
I do not believe that the official documentation is suitable for carefully explaining the disadvantages for users who do not need namespace packages.

My post: https://dev.to/methane/don-t-omit-init-py-3hga
@ncoghlan 's post: https://python-notes.curiousefficiency.org/en/latest/python_concepts/import_traps.html

>       Use :term:`regular package` always when it fits your needs.

Reviewer: ```suggestion
      It is recommended to use a :term:`regular package` when possible.
```

Author: Does "When possible" and "fits your needs" have same nuance?

For example, user who are thinking splitting their packages to two distribution, but it is "possible" to keep releasing one distribution.

In this case, regular package is not fits their need, but it is possible to use regular package.

Reviewer 2: The suggested edit makes text less direct, which is the style wanted for our docs.

Reviewer 2: But I think the wording should be improved. «always» seems strong but is weakened by «when it fits your needs» – people may not now if they do need to split a project!

Also, «across multiple directories» is talking about source trees but «across multiple distributions» is a concern for packaging and installation, so they’re not on the same level.

I’ll think on this and suggest edits!

Reviewer: I agree, it is friendlier

Reviewer 2: You misunderstand – I meant that the previous wording was more direct, which is what we want. «It is recommended» is not an improvement.

Reviewer 3: For context, @StanFromIreland, [Open telemetry](https://github.com/open-telemetry/opentelemetry-python) is an example of a namespace package.

I recommend removing "On the other hand ... your needs." This is a glossary entry so the prose should be brief and direct. I think if you add @methane's suggestion *Namespace packages allow several individually installable packages to have a common parent package.*, then there is no need for the two sentences and they can be removed.

      modules, respectively.

   namespace package
      A :term:`package` which serves only as a container for subpackages.
      Namespace packages may have no physical representation,
      and specifically are not like a :term:`regular package` because they
      have no ``__init__.py`` file.

      Namespace packages allow several individually installable packages to have a common parent package.

>       Otherwise, it is recommended to use a :term:`regular package`.

Reviewer 2: I might suggest this change:
```suggestion
      If you don't have this need, it is recommended to use a :term:`regular package`.
```

But I’m also ok with the proposed wording.

Reviewer 2: Ah, this goes back to wording that was omitted thanks to @willingc ’s review, so don’t apply my suggestion. We don’t want readers to ask themselves «do I have this need?!»
